cdap-api/src/main/java/co/cask/cdap/api/workflow/TokenValueWithTimestamp.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.workflow;
+
+/**
+ * Class representing the value for the key along with the timestamp inside the {@link WorkflowToken}.
+ */
+public final class TokenValueWithTimestamp {
```

Comment:
It is not the value of the token (token holds many key/values...). I think it is fine to name this just ValueWithTime